src/pages/api/logs/[id].js

```diff
@@ -0,0 +1,44 @@
+import mongoose from "mongoose";
+import { deleteLog } from "../../../../server/db/actions/Log";
+
+export default async function handler(req, res) {
+  if (req.method == "DELETE") {
+    if (!mongoose.isValidObjectId(req.query.id)) {
+      return res.status(422).send({
+        success: false,
+        message: "Unable to delete because log id is not in valid format",
+      });
+    }
+
+    return deleteLog(req.query.id)
+      .then((id) => {
+        return res.status(200).send({
+          success: true,
+          message: "Successfully deleted log",
+          data: { _id: id },
+        });
+      })
+      .catch((error) => {
+        if (
```

Comment:
move this check up to when we make the query for the log to delete. May have to refactor this block of code because we shouldn't rely on the node.js error message (changes with node versions). One possible solution is using error codes: https://nodejs.org/api/errors.html#errorcode